tools/disasm/disasm.py

```diff
@@ -94,62 +94,85 @@ def main():
     spimdisasm.mips.InstructionConfig.parseArgs(args)
     spimdisasm.common.GlobalConfig.parseArgs(args)
 
-    spimdisasm.common.GlobalConfig.ASM_USE_PRELUDE = False
     spimdisasm.common.GlobalConfig.PRODUCE_SYMBOLS_PLUS_OFFSET = True
     spimdisasm.common.GlobalConfig.TRUST_USER_FUNCTIONS = True
 
     dma_files = parse_file_addresses(args.config_dir / "file_addresses.csv")
 
-    output_files = collections.defaultdict(list)
+    print("Loading disasm info...")
+    all_file_splits: list[spimdisasm.mips.FileSplits] = []
     with open(args.rom, "rb") as f:
         for dma_file in dma_files:
             file_splits = load_file_splits(context, args.config_dir, dma_file, f)
+            all_file_splits.append(file_splits)
 
-            for section_type, files in file_splits.sectionsDict.items():
-                # TODO: disassemble overlay reloc sections?
-                if section_type == spimdisasm.common.FileSectionType.Reloc:
-                    continue
-
-                for path, section in files.items():
-                    output_files[path].append(section)
-
-    for path, sections in sorted(output_files.items()):
-        spimdisasm.common.Utils.printQuietless(f"Analyzing {path} ...")
-        for section in sections:
-            section.analyze()
-
-    for path, sections in sorted(output_files.items()):
-        spimdisasm.common.Utils.printQuietless(f"Writing {path} ...")
-        output_path = args.output_dir / path
-        output_path.parent.mkdir(parents=True, exist_ok=True)
-        with open(output_path, "w") as f:
-            f.write('.include "macro.inc"\n')
-            f.write("\n")
-            f.write(".set noat\n")
-            f.write(".set noreorder\n")
-            f.write(".set gp=64\n")
-            for section in sections:
-                f.write("\n")
-                f.write(f".section {section.sectionType.toStr()}\n")
-                f.write("\n")
-                f.write(f".align 4\n")
-                f.write("\n")
-                section.disassembleToFile(f)
-
-        if args.split_functions is not None:
-            rodata_list = []
-            for section in sections:
-                if section.sectionType == spimdisasm.common.FileSectionType.Rodata:
-                    rodata_list.append(section)
-
-            for section in sections:
-                if section.sectionType != spimdisasm.common.FileSectionType.Text:
-                    continue
-                output_dir = (args.split_functions / section.name).with_suffix("")
-                for func in section.symbolList:
-                    spimdisasm.mips.FilesHandlers.writeSplitedFunction(
-                        output_dir, func, rodata_list
+    print("Analyzing...")
+    for i, file_splits in enumerate(all_file_splits):
+        f = i / len(all_file_splits)
+        spimdisasm.common.Utils.printQuietless(
+            f"{f*100:3.0f}%", "Analyzing", file_splits.name, end="    \r"
```

Comment:
I agree the result could look better, but
- if a line wraps (which is more likely the longer the line) then `\r` brings the cursor back to the last wrap, not the start of the printed line. this is fine because having a terminal as narrow as having most/everyone line wrap is impractical anyway, but the longest line is really long and reasonable to wrap. If every line was as long as this longest line (due to padding with spaces) it'd wrap every line in an otherwise reasonable terminal width
- this is easily done in Python but I kinda don't want to do the same in the shell script anyway